pkg/query-service/app/queryBuilder/functions.go

```diff
@@ -0,0 +1,286 @@
+package queryBuilder
+
+import (
+	"math"
+	"sort"
+
+	v3 "go.signoz.io/signoz/pkg/query-service/model/v3"
+)
+
+// funcCutOffMin cuts off values below the threshold and replaces them with NaN
+func funcCutOffMin(result *v3.Result, threshold float64) *v3.Result {
+	for _, series := range result.Series {
+		for idx, point := range series.Points {
+			if point.Value < threshold {
+				point.Value = math.NaN()
+			}
+			series.Points[idx] = point
+		}
+	}
+	return result
+}
+
+// funcCutOffMax cuts off values above the threshold and replaces them with NaN
+func funcCutOffMax(result *v3.Result, threshold float64) *v3.Result {
+	for _, series := range result.Series {
+		for idx, point := range series.Points {
+			if point.Value > threshold {
+				point.Value = math.NaN()
+			}
+			series.Points[idx] = point
+		}
+	}
+	return result
+}
+
+// funcClampMin cuts off values below the threshold and replaces them with the threshold
+func funcClampMin(result *v3.Result, threshold float64) *v3.Result {
+	for _, series := range result.Series {
+		for idx, point := range series.Points {
+			if point.Value < threshold {
```

Comment:
if `point.Value=NaN`, this evaluates to false and hence the point is not used for clamping?